module/Activity/src/Activity/Service/Activity.php

```diff
@@ -197,9 +222,54 @@ public function reset(ActivityModel $activity)
      */
     public function disapprove(ActivityModel $activity)
     {
+        $this->permissionManageActivityOrException();
+
         $activity->setStatus(ActivityModel::STATUS_DISAPPROVED);
         $em = $this->getServiceManager()->get('Doctrine\ORM\EntityManager');
         $em->persist($activity);
         $em->flush();
     }
+
+    /*
+     * Some general ACL functions
+     */
+
+    /**
+     * Checks if the user is allowed to view the activity, and if not, throws an used permission exception
+     */
+    public function permissionViewActivityOrException()
```

Comment:
I don't think that we have to worry a lot about the ACL being too granular, because we configure it in one place. I do agree that approve and disapprove are closely related (it would actually be weird to split those), but I'm not so sure about reset there.

Anyway, I am not sure on my opinion on the second part. Indeed, it is true that including that check in the function itself isn't DRY. But neither is this solution (to be DRY here, you would have to combine everything into one function, but that would also clutter things in a different way).